### PR TITLE
ElasticSearch output coverage

### PR DESCRIPTION
**Pull request by sgomezg**
Elasticsearch test coverage. Removed unused functions and refactored others to improve readability or testability.

### CODE REVIEW

File: plugins/output-elasticsearch/src/test/scala/com/stratio/sparkta/plugin/output/elasticsearch/ElasticSearchOutputSpec.scala [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      "map".typed(FieldType.ObjectType),
+      "string" typed FieldType.StringType index "not_analyzed",
+      "binary".typed(FieldType.BinaryType)
+    )
+
+    def equals(a: TypedFieldDefinition, b: TypedFieldDefinition): Boolean = (a.name, a.`type`) ==(b.name, b.`type`)
+
+    @tailrec
+    final def equals(a: Seq[TypedFieldDefinition], b: Seq[TypedFieldDefinition]): Boolean = {
+      if (a.size >= 1 && b.size >= 1)
```
Comment (arinconstrio) [inline plugins/output-elasticsearch/src/test/scala/com/stratio/sparkta/plugin/output/elasticsearch/ElasticSearchOutputSpec.scala]: .size>=1 could  be replaced by nonEmpty

File: plugins/output-elasticsearch/src/test/scala/com/stratio/sparkta/plugin/output/elasticsearch/ElasticSearchOutputSpec.scala [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+class ElasticSearchOutputSpec extends FlatSpec with ShouldMatchers {
+
+  trait BaseValues {
+
+    val localhost = Seq(("localhost", 9200))
+    val ip = Seq(("127.0.0.1", 9200))
+    val remote = Seq(("dummy", 9300))
+    val properties = Map("nodes" -> new JsoneyString( """[{"node":"localhost","defaultPort":"9200"}]"""),
+      "dateType" -> "timestamp")
+    val mock = new ElasticSearchOutput("ES-out", properties, None, None)
```
Comment (arinconstrio) [inline plugins/output-elasticsearch/src/test/scala/com/stratio/sparkta/plugin/output/elasticsearch/ElasticSearchOutputSpec.scala]: I think that calling mock to the class under test is a little bit confused

File: plugins/output-elasticsearch/src/test/scala/com/stratio/sparkta/plugin/output/elasticsearch/dao/ElasticSearchDaoSpec.scala [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import org.scalatest.{FlatSpec, ShouldMatchers}
+
+import com.stratio.sparkta.sdk.TypeOp
+
+@RunWith(classOf[JUnitRunner])
+class ElasticSearchDAOSpec extends FlatSpec with ShouldMatchers {
+
+  trait BaseValues {
+
+    val mock = new ESDaoMock(Seq(("localhost", 9200)))
```
Comment (arinconstrio) [inline plugins/output-elasticsearch/src/test/scala/com/stratio/sparkta/plugin/output/elasticsearch/dao/ElasticSearchDaoSpec.scala]: I think that calling mock to the class under test is a little bit confused

File: plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/ElasticSearchOutput.scala [hunk 4]
```diff
@@ -47,74 +47,66 @@ class ElasticSearchOutput(keyName: String,
...
+      case IntegerType => structField.name typed FieldType.IntegerType
+      case BooleanType => structField.name typed FieldType.BooleanType
+      case DateType => structField.name typed FieldType.DateType
+      case TimestampType => structField.name typed FieldType.DateType
+      case ArrayType(_, _) => structField.name typed FieldType.MultiFieldType
+      case MapType(_, _, _) => structField.name typed FieldType.ObjectType
+      case StringType => structField.name typed FieldType.StringType index "not_analyzed"
       case _ => structField.name typed FieldType.BinaryType
     })
   }
```
Comment (Gschiavon) [inline plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/ElasticSearchOutput.scala]: Method getHostPortConfs() it's used just in ElasticSearchOutput so i think it shouldn't be in properties.

Comment (sgomezg) [inline plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/ElasticSearchOutput.scala]: Mongo format is something like ```"host:port,host:port"```. We can change ```MongoDbOutput.getConnentionConfs``` to something like this:

```
 override val nodes = properties.getHostPortConfs("nodes", DefaultNode, DefaultPort)

// Input (Seq(host,port),(host,port)), output "host:port,host:port"
val nodesFormatted = nodes.map(t => s"${t._1}:${t._2}").mkString(",")
```
What do you think?

File: plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/ElasticSearchOutput.scala [hunk 5]
```diff
@@ -47,74 +47,66 @@ class ElasticSearchOutput(keyName: String,
...
+  def indexNameType(tableName: String): String = (tableName + "/" + mappingType.get).toLowerCase
 
   override def setup: Unit = createIndices
 
-  private def createIndices = {
-    bcSchema.get.filter(tschema => (tschema.outputName == keyName)).foreach(tschemaFiltered => {
-      val tableSchemaTime = getTableSchemaFixedId(tschemaFiltered)
-      createIndexAccordingToSchema(tableSchemaTime.tableName, tableSchemaTime.schema)
-    })
+  def createIndices: Unit = {
```
Comment (danielcsant) [inline plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/ElasticSearchOutput.scala]: why do you remove private modifier?

Comment (sgomezg) [inline plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/ElasticSearchOutput.scala]: My mistake. I was testing mocks and I forgot undo not needed changes.

**Comment by sgomezg** [comment]
Thanks to everyone for the comments. All changes done.

**Pull request merged**